src/Model/ModelManager.php

```diff
@@ -207,17 +234,26 @@ public function batchDelete(string $class, BaseProxyQueryInterface $query): void
         \assert($iterator instanceof Iterator);
 
         $i = 0;
-        foreach ($iterator as $object) {
-            $documentManager->remove($object);
 
-            if (0 === (++$i % 20)) {
-                $documentManager->flush();
-                $documentManager->clear();
+        try {
+            foreach ($iterator as $object) {
+                $documentManager->remove($object);
+
+                if (0 === (++$i % 20)) {
+                    $documentManager->flush();
+                    $documentManager->clear();
+                }
             }
-        }
 
-        $documentManager->flush();
-        $documentManager->clear();
+            $documentManager->flush();
+            $documentManager->clear();
+        } catch (Exception|MongoDBException $exception) {
+            throw new ModelManagerException(
+                sprintf('Failed to delete object: %s', $class),
```

Comment:
IIUC, this exception is thrown if the removal of a single object fails, but it does not includes a reference for that object (its id or something else). Additionally, if the removal of a single object fails, the previous processed transactions are not reverted, so just mentioning the object class without a reference seems sub-optimal.

Reply:
Take a look at how it is implemented on the orm side: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/4.x/src/Model/ModelManager.php#L348

I just copied the same code and adapted for odm the exceptions.

Reply:
Maybe we should improve this message for both managers then.
WDYT?

Reply:
Yes it could be improved, but IMO it should be done in another PR. Here I am just trying to match behaviors of both persistence bundles and to avoid unexpected exceptions.

Also I am not familiar enough with MongoDB to know if there are transactions, and how to properly test that (I would do that other change you propose in another PR too).